server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -2199,7 +2199,7 @@ public void close() throws IOException {
         awaitPendingClose();
     }
 
-    private void awaitPendingClose() {
+    protected final void awaitPendingClose() {
```

Comment:
aren't we getting the closedLatch in the `closeNoLock` method parameter?

Reply:
Maybe it's helpful to describe how you're planning to use this method in the inheritors

Reply:
> aren't we getting the closedLatch in the `closeNoLock` method parameter?

Yes, but that one is final in InternalEngine

> Maybe it's helpful to describe how you're planning to use this method in the inheritors

Exposing the method as package-protected in implementations, allowing its usage in tests.

Reply:
See the linked pull request for such an usage.